Public Chat 뼈대 구현

peas856 opened the pull request:
### 주요 변경사항
---
1. showChat이라는 상태를 추가해서 참여가능한 채팅방 혹은 참여중인 채팅방을 클릭하면 {true, 클릭한 roomid}로 변경됩니다. true가 되면 참여 가능한 채팅방 리스트가 아닌 ChatList컴포넌트가 랜더링됩니다. 
2. messages 상태를 추가하여 key:roomId, val:chat[] 로 RECEIVE 이벤트로 받은 메시지들을 저장합니다. 추후 ChatList 컴포넌트로 chat[]을 넘겨줍니다. 
3. Notice 이벤트는 join과 leave를 전달하는데 senderUid가 로그인한 uid와 일치하면 참여중인방리스트 상태를 업데이트하는 부분을 추가했습니다. 
4. ChatList에 메시지 보내는 부분 구현이 안되어있어 일단 전송 버튼 클릭시 소켓으로 SEND이벤트를 보내 테스트해볼 예정입니다. 
### 이슈 및 문제
---
- 아직 테스트는 안되었으며 대강의 로직을 코드로 구현한 상태
- socket을 prop으로 계속 전달해주고 있는데 ContextAPI 혹은 Redux를 사용해야 하지 않을까 싶습니다. 
- 백엔드 오류 수정후 연결 및 테스트 해볼 예정

## 개요

<!--
이 PR이 해결한 이슈를 아래처럼 추가해주세요.
resolved와 같은 키워드를 **반드시** 포함해야 이 PR이 머지될 때 연관된 이슈도 같이 닫힙니다.
참고: https://bumkeyy.gitbook.io/bumkeyy-code/project-management/pull-request

- closed #15
- resolved #16
-->

Commit:
chat 수정중

Commit:
Merge branch 'develop' into chat

Commit:
feat: Chat 컴포넌트 연결 중(Public chat)

scarf005 commented:
지금 완성된 PR인가요?
작성중인 PR이라면 draft (작업 중)로 전환해주시면 좋을 것 같아요
reviewer 항목 밑에 있습니다

peas856 commented:
전환했습니다! 작업 진행상황 공유 및 유킴님 의견을 묻기 위한 PR이었습니다.

Commit:
Merge branch 'develop' into chat

Commit:
feat: 실시간 채팅 기능 구현 및 방 만들기 및 방 참여하기 구현 초기버전

scarf005 commented:
> ChatList에 메시지 보내는 부분 구현이 안되어있어 일단 전송 버튼 클릭시 소켓으로 SEND이벤트를 보내 테스트해볼 예정입니다.
> 아직 테스트는 안되었으며 대강의 로직을 코드로 구현한 상태

혹시 이 부분은 어떻게 하셨나요? 변경점 확인할 수 있을까요?

hoikii committed:
join이벤트 인자 수정

#125 때문에
```
prop.socket.emit('JOIN', room)
```
를
```
prop.setShowChat({ bool: true, roomId: room })
```로 고쳐야 할 것 같습니다

peas856 commented:
1. ChatList 컴포넌트에 socket과 roomId 를 prop으로 받아 임시로 만든 버튼으 ㄹ누르면 아래 함수가 실행되어 SEND이벤트를 발생시킵니다. 
`const sendmsg = () => {
    socket.emit('SEND', {
      roomId: roomId,
      msgContent: 'what do you say?',
      createdAt: new Date(),
    })
  }`
2. 그럼 같은 방에 있는 유저는 RECEIVE 이벤트를 받아 ChatView의 Messages가 업데이트 되어 받은 메시지를 표시합니다. 
3. 다만 같은 메시지가 2번가는 버그가 있는데 임시로 구현한 버튼이라 일단 Input 컴포넌트 구현되면 합쳐서 버그 수정 예정입니다. 
4. 또한 ChatListItem에 User부분이 미구현되어 있어 언노운 유저로 표시되고 있습니다. 
5. 임시로 구현한거라 Input 구현하시면 버튼 지우고 Send 이벤트 발생시키면 될 것 같습니다.

scarf005 committed:
Merge branch 'develop' into chat

scarf005 reviewed (commented):
이제 입력 기능 추가하면 되겠네요 수고 많으셨습니다!
